cmd/otel-allocator/README.md

```diff
@@ -4,6 +4,136 @@ The TargetAllocator is an optional separately deployed component of an OpenTelem
 distribute targets of the PrometheusReceiver on all deployed Collector instances. The release version matches the
 operator's most recent release as well.
 
+In essence, Prometheus Receiver configs are "stolen" and replaced with a http_sd_configs directive that points to the 
+Allocator, these are then loadbalanced/sharded to the collectors. In addition, the TargetAllocator can discover targets
+via Prometheus CRs (currently ServiceMonitor, PodMonitor) which it presents as scrape configs and jobs on the
+`/scrape_configs` and `/jobs` endpoints respectively.
+
+# Usage
+The `targetAllocator:` controls the TargetAllocator general properties. Full API spec can be found here: [api.md#opentelemetrycollectorspectargetallocator](../../docs/api.md#opentelemetrycollectorspectargetallocator)
+
+A basic example that deploys, but does nothing.
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: collector-with-ta
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [prometheus]
+          processors: []
+          exporters: [logging]
+```
+
+## PrometheusCR specifics
+TargetAllocator discovery of PrometheusCRs can be turned on with
+```yaml
+  targetAllocator:
+    enabled: true
+    prometheusCR:
+      enabled: true
+```
```

Comment:
Maybe instead of a YAML snippet we could just specify this can be enabled with `prometheusCR.enabled=true` and link to docs for additional features?

Reply:
Sure, with a full path, I think that is OK.
The reason I did it this way is to show the the targetAllocator has to be enabled - something I can totally see someone leaving out.

Reply:
Changed to full path.